src/openforms/appointments/api/views.py

```diff
@@ -167,3 +181,46 @@ def get_objects(self):
         client = get_client()
         times = client.get_times([product], location, serializer.validated_data["date"])
         return [{"time": time} for time in times]
+
+
+@extend_schema(
+    summary=_("Cancel an appointment"),
+    responses={
+        204: None,
+        403: OpenApiResponse(
+            response=ExceptionSerializer,
+            description=_("Unable to verify ownership of the appointment."),
+        ),
+        502: OpenApiResponse(
+            response=ExceptionSerializer,
+            description=_("Unable to cancel appointment."),
+        ),
+    },
+)
+class CancelAppointmentView(GenericAPIView):
+    lookup_field = "uuid"
+    lookup_url_kwarg = "submission_uuid"
+    queryset = Submission.objects.all()
+    authentication_classes = ()
+    permission_classes = [ActiveSubmissionPermission]
+
+    def post(self, request, *args, **kwargs):
+        submission = self.get_object()
+
+        serializer = CancelAppointmentInputSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        emails = submission.get_email_confirmation_recipients(submission.data)
+
+        if serializer.validated_data["email"] not in emails:
+            raise PermissionDenied
```

Comment:
A comment describing the business logic/requirement from the ticket here would provide context that is now completely absent.